### DISCUSSION TITLE
Legendary.nvim Configs

### DISCUSSION THREAD

**Discussion by mrjones2014** [mrjones2014/legendary.nvim#95] [Show and tell]
If you want to show off your `legendary.nvim` configs, do it here!

- [My `legendary.nvim` configuration](https://github.com/mrjones2014/dotfiles/blob/master/.config/nvim/lua/my/configure/mine/legendary.lua)
  - Which imports my [keymap](https://github.com/mrjones2014/dotfiles/blob/master/.config/nvim/lua/my/legendary/keymap.lua), [commands](https://github.com/mrjones2014/dotfiles/blob/master/.config/nvim/lua/my/legendary/commands.lua), [autocmds](https://github.com/mrjones2014/dotfiles/blob/master/.config/nvim/lua/my/legendary/autocmds.lua), and [functions](https://github.com/mrjones2014/dotfiles/blob/master/.config/nvim/lua/my/legendary/functions.lua) modules
  - The `lsp_*` maps in the above modules only get bound in my [LSP `on_attach` function](https://github.com/mrjones2014/dotfiles/blob/master/.config/nvim/lua/my/lsp/utils/init.lua)

**Comment by nikfp** [comment]
Hello! I'm hitting a 404 with your links, and I'd like to see how to set this up within LSP on_attach functions for language dependent mappings. Any insight? 

Thanks for building this BTW.

**Reply by mrjones2014** [reply]
Sorry about that! I moved some stuff around in my config since I created this discussion. Links are updated now.

**Reply by nikfp** [reply]
Thanks for making that change! Now I can spend some time figuring out how you did what you did. I'm new-ish to Nvim and Lua both, and you are clearly not. Looks like some great patterns I can learn from while I sort through this. I love how clean everything is.

**Reply by mrjones2014** [reply]
Let me know if you have questions 🙂

**Comment by zaatas** [comment]
Re: Load keymaps, etc. before calling require('legendary').keymaps(keymaps/commands, etc)

Question: Would it be possible, to use a _G to queue keymaps/commands, etc?  Then via an autocmd call require at VimEnter ?

Example: _G.KM.add( {'<leader><c-p>', :Legendary<CR>', description='Legendary Find', opt = myOpts}, ... )

This way I can map keys/commands anywhere in my config sequence, even before Legendary has loaded.  Which is want I want to do.  

Many thanks / Matt R.

**Reply by mrjones2014** [reply]
There isn’t any automatic reading from globals or anything but you could easily implement it yourself in user config:

```lua
vim.g.legendary_keymaps = {}

-- do some stuff, add stuff to vim.g.legendary_keymaps

-- do more stuff

-- some time later 
require('legendary').setup({
  keymaps = vim.g.legendary_keymaps,
})
```

**Reply by zaatas** [reply]
Yep.. I added:

`_G.ZA.keys = {}`
`_G.ZA.cmds = {}`

-- legend insert keymap
`_G.LegK= function(keymap)
  table.insert(ZA.keys, keymap)
end `

-- legend insert commands
`_G.LegC = function(command)
  table.insert(ZA.cmds, command)
end `

Example:
  `LegK( {"<Leader>fe", "<cmd>Telescope file_browser <CR>", description="Telescope File Browser", ZA.dopt } )`


`  au VimEnter * lua require('legendary').commands(ZA.cmds)`
 ` au VimEnter * lua require('legendary').keymaps(ZA.keys)`

This works fine.  

Note: 
- When I do a cold install of nvim/init.lua.  Legendary is not installed.
- I do not centralize keymaps.  Maps are created at the config site.
- So this helps to wait... 

I use Legendary 100% to map keys/cmds/autocmd all over my config -- amazing plug you have created!!!.

**Reply by mrjones2014** [reply]
Glad you like it! You could do it with a single autocmd by using `setup()`:

```lua
au VimEnter * lua require('legendary').setup({ keymaps = ZA.keys, commands = ZA.cmds  })
```

**Reply by zaatas** [reply]
10,000 thanks.. I have implemented it.
